### PR TITLE
Nir zalcman/jooq service pet

### PR DESCRIPTION
**Pull request by nirzalcman**
Pet service and pet service test for the jooq task.

### CODE REVIEW

File: src/main/kotlin/com/hibob/academy/service/OwnerService.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/service/OwnerService.kt]: Why all this enters?

File: src/main/kotlin/com/hibob/academy/resource/PetResource.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        return Response.ok(petService.updatePetOwner(id,ownerId,companyId)).build()
+    }
+
+
+
+
+
+
+
+}
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/resource/PetResource.kt]: Why all these enters?

File: src/main/kotlin/com/hibob/academy/service/OwnerService.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        return ownerDao.getOwnerById(id, companyId)
+    }
+
+    fun getOwnerByPetId(petId: Long, companyId: Long): Owner?{
+        val owner = ownerDao.getOwnerByPetId(petId, companyId)
+        when{
+            owner ==null -> throw NotFoundException("Pet Not Found")
+            owner.id ==null  -> throw NotFoundException("The pet dont has owner")
+            else -> return owner
+        }
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/service/OwnerService.kt]: Cant you use elvis operator?

File: src/main/kotlin/com/hibob/academy/service/PetService.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    fun getPetsByType(type: String, companyId: Long): List<Pet> {
+        return petDao.getPetsByType(type, companyId)
+    }
+
+    fun getPetById(id: Long, companyId: Long): Pet? {
+        return petDao.getPetById(id, companyId)
+    }
+
+    fun updatePetOwner(petId: Long, ownerId: Long, companyId: Long): Int {
+        return if (petDao.updatePetOwner(petId, ownerId, companyId) > 0) 1 else throw NotFoundException("Pet Not Found")
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/service/PetService.kt]: Do we want to return Int or Boolean if success or maybe even doesnt return anything only ok response?

File: src/main/kotlin/com/hibob/academy/service/OwnerService.kt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        return ownerDao.getOwners(companyId)
+    }
+
+    fun createOwner(ownerRequest: OwnerCreationRequest): Long {
+        return ownerDao.createOwner(ownerRequest)
+
+    }
+
+    fun getOwnerById(id: Long, companyId: Long): Owner? {
+        return ownerDao.getOwnerById(id, companyId)
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/service/OwnerService.kt]: Do we want to return `Owner?` or `Owner `?
Why not handle the null here?

File: src/main/kotlin/com/hibob/academy/service/PetService.kt [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+    fun createPet(petCreationRequest: PetCreationRequest): Long {
+        return petDao.createPet(petCreationRequest)
+    }
+
+    fun getPetsByType(type: String, companyId: Long): List<Pet> {
+        return petDao.getPetsByType(type, companyId)
+    }
+
+    fun getPetById(id: Long, companyId: Long): Pet? {
```
Comment (omermelamed) [inline src/main/kotlin/com/hibob/academy/service/PetService.kt]: Do we want to return `Pet?` or `Pet` ?
Why not handle the `null` here?

Comment (nirzalcman) [reply src/main/kotlin/com/hibob/academy/service/PetService.kt]: Yes its better i changed it

File: src/main/kotlin/com/hibob/academy/service/PetService.kt (hunk 4 above)
Comment (nirzalcman) [reply src/main/kotlin/com/hibob/academy/service/PetService.kt]: I think return true its the better option so i changed to this

**Review by omermelamed — APPROVED** [review]

**Pull request merged**